################################################################################
SHORT DESCRIPTION: 
################################################################################
Trace the TPM TCG vendor ID.

################################################################################
LONG DESCRIPTION: 
################################################################################
Output the TPM TCG vendor ID with the device ID and the revision ID to identify
the TPM device in use on the log output.

################################################################################
CHANGELOG 
################################################################################
Original author: Ross Philipson <philipsonr@ainfosec.com>
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
This patch is for debugging purposes and can be safely removed.

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES 
################################################################################
Index: linux-4.1.6/drivers/char/tpm/tpm_tis.c
===================================================================
--- linux-4.1.6.orig/drivers/char/tpm/tpm_tis.c	2015-09-11 15:18:57.707259988 +0200
+++ linux-4.1.6/drivers/char/tpm/tpm_tis.c	2015-09-11 15:19:36.046739133 +0200
@@ -649,9 +649,10 @@
 	vendor = ioread32(chip->vendor.iobase + TPM_DID_VID(0));
 	chip->vendor.manufacturer_id = vendor;
 
-	dev_info(dev, "%s TPM (device-id 0x%X, rev-id %d)\n",
+	dev_info(dev, "%s TPM (vendor-id 0x%X device-id 0x%X, rev-id %d)\n",
 		 (chip->flags & TPM_CHIP_FLAG_TPM2) ? "2.0" : "1.2",
-		 vendor >> 16, ioread8(chip->vendor.iobase + TPM_RID(0)));
+		 vendor & 0xffff, vendor >> 16,
+                 ioread8(chip->vendor.iobase + TPM_RID(0)));
 
 	if (!itpm) {
 		probe = probe_itpm(chip);
